### DISCUSSION TITLE
headlines.nvim headline symbols don't show

### DISCUSSION THREAD

**Discussion by kapral18** [LazyVim/LazyVim#1816] [Q&A] [answered]
How do you fix this?

![CleanShot 2023-10-20 at 12 02 37@2x](https://github.com/LazyVim/LazyVim/assets/1625373/46086683-294f-44aa-ada5-199130cd2447)

In the screenshot you can see the symbols of markdown headlines not rendering properly

**Comment by dpetka2001** [comment]
Could you elaborate on what you're expecting to see, because I fail to understand?

**Reply by kapral18** [reply]
The markdown headlines from headlines.nvim( I assume) are not rendering properly. I updated the description.

**Reply by folke** [reply]
Your colorscheme has to support the headline hl groups

**Reply by kapral18** [reply]
@folke  so when you added this functionality to the markdown extra did you test it with a specific colorscheme that supported it? Can you share it? Because I just ran through every single colorscheme I have and not a single one has a proper headlines support apparently...

**Comment by abeldekat** [comment]
Also worth noting:

`nvim README.md`

The plugin is loaded but headlines do not appear...

Open the file from within `Neovim`: The headlines appear, as shown by @kapral18.
However, when changing color scheme, Tokyo night storm into moon, some headlines *temporarily* change into very white. Moving the cursor restores the headlines.

**Reply by folke** [reply]
That's probably a bug in headlines.nvim where they don't reload on colorscheme change

**Reply by dpetka2001** [reply]
@folke I use Tokyonight by default, which supports the `Headline1/2/3` groups but I can confirm that opening directly a file with `nvim README.md` doesn't show headlines, whereas opening Neovim with `nvim` and then `find_file` or `restore_session` does show the headlines as intended.

**Reply by abeldekat** [reply]
Adding the following to the opts removes the squares:

```lua
{
    "lukas-reineke/headlines.nvim",
    opts = function(_, opts)
      opts.markdown = {
        fat_headline_lower_string = "▔",
      }
      return opts
    end,
}
```

See this [comment](https://github.com/lukas-reineke/headlines.nvim/issues/41#issuecomment-1556334775)

**Reply by dpetka2001** [reply]
But it still doesn't work when passing the file arguments to `nvim`, right? Only opening the file from inside nvim. Coincidentally, I also use `Fira Code` and that must have been why I didn't see any squares in my case.

**Reply by abeldekat** [reply]
I am using jetbrains-mono...)

Indeed, when passing file arguments, the headlines still do not show.
Considering that the plugin loads correctly when opening a file from inside Neovim, I think the problem is not the plugin itself. 
I intend to test this scenario with older versions of LazyVim/lazy.nvim.

**Reply by dpetka2001** [reply]
It has something to do with how `ft` events are handled I believe. I tried with following config
```lua
return {
  {
    "lukas-reineke/headlines.nvim",
    event = "VeryLazy",
    ft = function()
      return {}
    end,
    opts = function(_, opts)
      opts.markdown = {
        fat_headline_lower_string = "▔",
      }
      return opts
    end,
  },
}
```
and it correctly shows the headlines both with and without file arguments.

**Reply by abeldekat** [reply]
> I intend to test this scenario with older versions of LazyVim/lazy.nvim.

Thanks @dpetka2001!

@folke, perhaps this problem is related to `lazy.nvim`?

**Reply by dpetka2001** [reply]
I just tried different Neovim versions and in nightly it works correctly with the initial `ft` event setup. It's just in 0.9.4 stable that the initial event setup does not seem to work correctly. Could someone else confirm the difference between nightly and stable?

**Reply by folke** [reply]
I'll look into it tomorrow. Feel free to submit an issue linked here so I dont forget

**Comment by folke** [comment, marked as answer]
There's three different things happening here:

1. op's intial issue: that's really just the font. Nothing I can do about that. Headlines uses regular box drawing symbols that any font should support
2. I configured headlines to use numbered Headline groups. Those are supported in TokyoNight, but not in other colorschemes, resulting in those ugly white lines. Just pushed a fix for that that adds default groups like `Headline1` => `Headline`
3. There seems to be an issue with `lazy_file` I'm going to look at. (re: `nvim README.md`)

**Comment by abeldekat** [comment]
Folke fixed the third issue as well, see this [comment](https://github.com/LazyVim/LazyVim/pull/1858#issuecomment-1779308477).

**Comment by aaharr** [comment]
Just encountered this issue with Fira Code Nerd Font, today's one true font. The issue for me is that it is missing the upper 3/8 block but not the lower 3/8 block. One solution is to replace with half blocks in opts, or with empty strings if you don't need fat headlines.
```
                    -- Fira code has lower 3/8 block character, but no upper 3/8 block character
                    -- Replace with half blocks
                    fat_headline_upper_string = "▄",
                    fat_headline_lower_string = "▀",
```
Hope this helps, future self >.>

**Reply by verisimillipede** [reply]
Works perfectly, thank you very much
